lib/v2/screens/explore_screens/invite/invite_screen.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:seeds/v2/blocs/rates/viewmodels/rates_bloc.dart';
+import 'package:seeds/v2/components/alert_input_value.dart';
+import 'package:seeds/v2/components/amount_entry_widget.dart';
+import 'package:seeds/v2/components/balance_row.dart';
+import 'package:seeds/v2/components/flat_button_long.dart';
+import 'package:seeds/v2/components/full_page_error_indicator.dart';
+import 'package:seeds/v2/components/full_page_loading_indicator.dart';
+import 'package:seeds/v2/domain-shared/page_state.dart';
+import 'package:seeds/v2/domain-shared/ui_constants.dart';
+import 'package:seeds/v2/design/app_theme.dart';
+import 'package:seeds/v2/screens/explore_screens/invite/interactor/viewmodels/bloc.dart';
+
+/// INVITE SCREEN
+class InviteScreen extends StatelessWidget {
+  const InviteScreen({Key? key}) : super(key: key);
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+      create: (context) => InviteBloc(BlocProvider.of<RatesBloc>(context).state)..add(const LoadUserBalance()),
+      child: Scaffold(
+        appBar: AppBar(title: Text('Invite', style: Theme.of(context).textTheme.headline7)),
+        body: BlocConsumer<InviteBloc, InviteState>(
+          listenWhen: (_, current) => current.showInviteLinkDialog,
+          listener: (context, state) {
+            // Here will go a invite link dialog
+          },
+          builder: (context, InviteState state) {
+            switch (state.pageState) {
+              case PageState.initial:
+                return const SizedBox.shrink();
+              case PageState.loading:
+                return const FullPageLoadingIndicator();
+              case PageState.failure:
+                return const FullPageErrorIndicator();
```

Comment:
Should we pass the error or error message to `FullPageErrorIndicator` to show a more relevant error message instead of a generic one? something to think about for the future maybe.